packages/api/cms-api/src/dam/dam.module.ts

```diff
@@ -40,7 +41,7 @@ interface DamModuleAsyncOptions extends Pick<ModuleMetadata, "imports"> {
 @Global()
 @Module({})
 export class DamModule {
-    static registerAsync(options: DamModuleAsyncOptions): DynamicModule {
+    static registerAsync(options: DamModuleAsyncOptions, Scope?: Type<DamScopeInterface>): DynamicModule {
```

Comment:
I can't add the Scope type to the DamModuleOptions because this would make the resolver generation async, which doesn't work with the GraphQL schema generation.

Reply:
Coudn't you change this to a non-async `register`?

Reply:
This might work because of our new approach to "injecting" the config. Do we plan to require this new approach in every application?

Reply:
> This might work because of our new approach to "injecting" the config.

I don't think that was async?

> Do we plan to require this new approach in every application?

I would say so, the only argument against ist that upgrading is more work.

Reply:
> I don't think that was async?

Thereby I meant that the old module registration was async due to the injection of the ConfigModule. But we don't require this injection anymore, so we may make the registration sync. Done in 6c19eaf410ee833bed83cbe16e62ef8d1104ad5a.